tests/roles/dataplane_adoption/tasks/main.yaml

```diff
@@ -385,6 +385,57 @@
         echo "Run out of retries"
         exit 2
 
+    - name: Create OpenStackDataPlaneDeployment to run the reboot check
+      when: run_reboot_check|bool
+      no_log: "{{ use_no_log }}"
+      ansible.builtin.shell: |
+        {{ shell_header }}
+        {{ oc_header }}
+        oc apply -f - <<EOF
+        apiVersion: dataplane.openstack.org/v1beta1
+        kind: OpenStackDataPlaneDeployment
+        metadata:
+          name: openstack-pre-adoption-reboot-check
+        spec:
+          nodeSets:
+          - openstack
+          servicesOverride:
+          - reboot-os
+          ansibleExtraVars:
+          - edpm_reboot_strategy: never
```

Comment:
does this cause the ansible code to fail if reboot is required?  If not then we need some additional logic below that stops the automated process if reboot is required.